### PR TITLE
Drastically improve coverage

### PR DESCRIPTION
**Pull request by ikeydoherty**
- Add a shedload of mocking
 - Spend 2 weeks lying to clr-boot-manager until it doesn't know what a real disk is anymore

Notable improvements:

 - We finally have coverage of syslinux!
 - Fixed some super nasty segfaults
 - Reduced a lot of complexity, by introducing complexity. ...

Notable absences:

 - `UUID` vs `PartUUID` support. We mock, and not fake, using actual mickeymouse `devfs` and `sysfs` trees. Thus we can't then properly trick a `UUID` without supporting non-GPT. To support this is outside the scope of this PR IMHO. This should be a legacy boot enabling PR.
 - LVM on LUKS. Again, out of scope, and we only pretended to have it.
 - Config file validation. We didn't have this before, and the pull request is already massive. We already have far more coverage to add and these will be added in follow up pull request to restore sanity for v2.0 cut.

### CODE REVIEW

**Comment by ikeydoherty** [comment]
15 = "drastic". :(

**Comment by bryteise** [comment]
Well geat progress at least. Also likely the biggest portion of testing rework is at least done. Off to review.

**Comment by ikeydoherty** [comment]
Thanks :D

File: src/lib/system_stub.c [hunk 1]
```diff
@@ -16,16 +16,37 @@
...
+ * This is the default internal function
+ */
+static char *cbm_devnode_to_devpath(dev_t dev)
+{
+        if (major(dev) == 0) {
+                LOG_ERROR("Invalid block device: %u:%u", major(dev), minor(dev));
+                return NULL;
+        }
+
+        autofree(char) *c = NULL;
```
Comment (bryteise) [inline src/lib/system_stub.c]: Little confused at first since this wasn't at the top of the function =P.

**Comment by bryteise** [comment]
Looking at this a little, I think adding a OOM macro for asprintf would likely increase coverage quite a bit (and make reading a little nicer too).

**Comment by ikeydoherty** [comment]
Should we merge a `string_or_die`, ala `swupd-client` ? Also, thanks for merging, and sorry for that horrific review :D

**Comment by bryteise** [comment]
I think that'd be a good readability win and hopefully a little coverage bump. No problem. The missing coverage doesn't look too scary now based on the functions/lines missing testing, good effort paying off!

**Comment by ikeydoherty** [comment]
Done xD

**Pull request merged**